### PR TITLE
Explicitly handle conflicts between IonPropertyName and field names #29

### PR DESCRIPTION
**Pull request by amzn-paunort**
*Issue #, if available:* #29

*Description of changes:*

In order to ensure that `IonPropertyName` is respected over any conflicting field names, this PR adds the `IonPropertyName` value to the list of serialized fields, so that any included fields with the same name are ignored when read.

It appears that the produced object already had this behaviour, since the nested call to `IonSerializer#Serialize` on the field appears later than the property and would not override the previous written value. However, this seemed more like an unintentional success, rather than an explicit handling of this edge case, so I have added the check explicitly here, as well as a unit test for the behaviour.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: Amazon.IonObjectMapper.Test/IonObjectSerializerTest.cs [hunk 1]
```diff
@@ -850,5 +850,23 @@ public void DeserializeToPrivateProperty()
...
+                firstString = "ignored",
+                SomeOtherFirstString = "expected"
+            });
+
+            var serialized = StreamToIonValue(stream);
+            Assert.IsFalse(serialized.ContainsField("SomeOtherFirstString"));
+            Assert.IsTrue(serialized.ContainsField("firstString"));
+            Assert.AreEqual(serialized.GetField("firstString").StringValue, "expected");
+
+            var deserialized = defaultSerializer.Deserialize<PropertyNameConflictsWithIonPropertyField>(IonReaderBuilder.Build(serialized));
```
Comment (guyilin-amazon) [inline Amazon.IonObjectMapper.Test/IonObjectSerializerTest.cs:868]: could be `defaultSerializer.Deserialize<PropertyNameConflictsWithIonPropertyField>(stream);`

**Review by guyilin-amazon — APPROVED** [review]
